richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java

```diff
@@ -0,0 +1,95 @@
+package org.fxmisc.richtext.skin;
+
+import com.sun.javafx.css.converters.PaintConverter;
+import javafx.beans.property.ObjectProperty;
+import javafx.css.CssMetaData;
+import javafx.css.Styleable;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.text.Text;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextExt extends Text {
+
+    private ObjectProperty<Paint[]> backgroundColor = null;
```

Comment:
Ranming it might cause confusion for the user who expects -fx-background-color.
So we can either just keep it like it is and tolerate the unnecessary Paint array. Or we rename it.
-fx-background-fill would be a good choice since it follows -fx-highlight-fill. It's your call.